metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_entities.py

```diff
@@ -0,0 +1,477 @@
+        self.set_datasets()
+
+    def get_datasets(self) -> Deque[DremioDataset]:
+        self.set_datasets()
+        return self.datasets
+
+    def set_containers(self) -> None:
+        if not self.containers_populated:
+            for container in self.dremio_api.get_all_containers():
+                container_type = container.get("container_type")
+                if container_type == DremioEntityContainerType.SOURCE:
+                    self.sources.append(
+                        DremioSource(
+                            container_name=container.get("name"),
+                            location_id=container.get("id"),
+                            path=[],
+                            api_operations=self.dremio_api,
+                            dremio_source_type=container.get("source_type"),
+                            root_path=container.get("root_path"),
+                            database_name=container.get("database_name"),
+                        )
+                    )
+                elif container_type == DremioEntityContainerType.SPACE:
+                    self.spaces.append(
+                        DremioSpace(
+                            container_name=container.get("name"),
+                            location_id=container.get("id"),
+                            path=[],
+                            api_operations=self.dremio_api,
+                        )
+                    )
+                elif container_type == DremioEntityContainerType.FOLDER:
+                    self.folders.append(
+                        DremioFolder(
+                            container_name=container.get("name"),
+                            location_id=container.get("id"),
+                            path=container.get("path"),
+                            api_operations=self.dremio_api,
+                        )
+                    )
+                else:
+                    self.spaces.append(
+                        DremioSpace(
+                            container_name=container.get("name"),
+                            location_id=container.get("id"),
+                            path=[],
+                            api_operations=self.dremio_api,
+                        )
+                    )
+
+        logging.info("Containers retrieved from source")
+
+        self.containers_populated = True
+
+    def force_reset_containers(self) -> None:
+        self.containers_populated = False
+        self.set_containers()
+
+    def get_containers(self) -> Deque:
+        self.set_containers()
+        return deque(itertools.chain(self.sources, self.spaces, self.folders))
+
+    def get_sources(self) -> Deque[DremioSource]:
+        self.set_containers()
+        return self.sources
+
+    def get_glossary_terms(self) -> Deque[DremioGlossaryTerm]:
+        self.set_datasets()
+        self.set_containers()
+        return self.glossary_terms
+
+    def force_set_glossary_terms(self) -> None:

```

Comment:
Is this or other methods used within used anywhere ?

Reply:
Removed that too